### DISCUSSION TITLE
Issue Cards Not Rendering Emojis nor MD Formatting

### DISCUSSION THREAD

**Discussion by tajmone** [community/community#23393] [New to GitHub]
<p>While using GH Community I noticed that the Issue cards generated from links to Issues on <a href="http://GitHub.com" rel="noopener nofollow ugc">GitHub.com</a> repositories are not rendering emoji inside labels:</p>
<aside class="onebox githubissue" data-onebox-src="https://github.com/fantaisie-software/purebasic/issues/144#issuecomment-785233553">
  <header class="source">

      <a href="https://github.com/fantaisie-software/purebasic/issues/144#issuecomment-785233553" target="_blank" rel="noopener nofollow ugc">github.com/fantaisie-software/purebasic</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/fantaisie-software/purebasic/issues/144#issuecomment-785233553" target="_blank" rel="noopener nofollow ugc">CONTRIBUTING.md : Git(hub) mini guide don't works</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-02-23" data-time="22:28:27" data-timezone="UTC">10:28PM - 23 Feb 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/Naheulf" target="_blank" rel="noopener nofollow ugc">
          <img alt="Naheulf" src="https://user-images.githubusercontent.com/11064292/181076720-203828e5-f8dd-4725-a823-ab98581b4683.png" class="onebox-avatar-inline" width="20" height="20">
          Naheulf
        </a>
      </div>
    </div>

    <div class="labels">
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">:books: documentation</span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">:hammer: Git</span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">:hammer: GitHub</span>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">### Description of the Issue
The steps to setup the fork  for contribution don'<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">t works with a new fork and git up to date.

### Steps to Reproduce the Issue

1. Fork fantaisie-software/purebasic from the GitHub website
2. open Git Bash
3. Clone the repo with `git clone https://github.com/&lt;github_username&gt;/purebasic.git`
(Replace `&lt;github_username&gt;` with your own github username.)
4. Move to the repo usin `cd purebasic`
5. Follow steps from [CONTRIBUTING.md#setting-up-your-fork-for-contributions](https://github.com/fantaisie-software/purebasic/blob/c3bca303f9e66a4574b62e775eaf1b78ba4468aa/CONTRIBUTING.md#setting-up-your-fork-for-contributions)

### Expected Behavior
Should works.

### Actual Behavior 1
The second command generate the error: `fatal: 'upstream/devel' is not a commit and a branch 'devel' cannot be created from it`

### Fix 1
Run: `git fetch upstream` after the first command.

### Actual Behavior 2
After use the fix 1 the second command generate another error: `fatal: A branch named 'devel' already exists.`

### Fix 2
On the second command replace the `-b` (lower case B) with `-B` (uppercase B)

### More info
I let more experienced users check if theses fix are good or tricky. And I let english native speaker update the CONTRIBUTING.md file.</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>As you can see in the above card, the emoji in the labels are rendered as text. Also, markdown formatting is not rendered either.</p>
<p>I’m not sure how and where these cards are generated (i.e. whether they are generated by GitHub or a third party service) and I know that since they are intended for any type of external link they don’t necessarily abide to GitHub convention.</p>
<p>But, in case they are being generated by GitHub, it would make sense that their generator would be context aware regarding URLs pertaining to GitHub Issues, PRs, Discussions, etc., and render labels, emoji and markdown formatting accordingly.</p>
<p>You might wonder why I prefix every label with an emoji icon. The answer is that they allow grouping related labels by topic, using a common emoji (e.g. a :skull: for problematic issues, and :hammer: for tools, a :books: for documentation, etc.). Since emojis are indexed as text, they are sorted Asciibetically, which ultimately allows labels to be listed as if they were in groups.</p>
<p>Also, emoji icons allow to quickly spot labels by type, without necessary having to read them.</p>

**Comment by Gisgar3** [comment, marked as answer]
<p>Hey there, welcome back to the <strong>GitHub Support Community</strong>! I’m fairly certain these cards may be generated by Discourse (the third-party service that GitHub uses for this forum) with some help of custom GitHub configuration in the backend. I do agree that emojis and Markdown should be rendered in these cards when referred to in forum threads.</p>
<p>I would typically refer you to the official feedback form on GitHub, but that only refers to GitHub services and not the support community forum. We appreciate the feedback and I’ll relay this thread to staff! Thank you :grin:</p>

**Comment by nyahbhinghiPrincess** [comment]
<p>Hi <a class="mention" href="/tajmone">@tajmone</a> and <a class="mention" href="/Gisgar3">@Gisgar3</a> I thank you both some much for surfacing and discussing this pattern!  We’ll take a look into it.</p>

**Comment by nyahbhinghiPrincess** [comment]
<p>Hi <a class="mention" href="/tajmone">@tajmone</a> <em>New</em> posts with purebasic emojis render successfully now from my view. We checked in with our platform support and they’ve pushed some new changes regarding this. Would be able to tell me if you see the same result?</p>
<aside class="onebox githubissue" data-onebox-src="https://github.com/fantaisie-software/purebasic/issues/144#issuecomment-785233553">
  <header class="source">

      <a href="https://github.com/fantaisie-software/purebasic/issues/144#issuecomment-785233553" target="_blank" rel="noopener">github.com/fantaisie-software/purebasic</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/fantaisie-software/purebasic/issues/144#issuecomment-785233553" target="_blank" rel="noopener">CONTRIBUTING.md : Git(hub) mini guide don't works</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-02-23" data-time="22:28:27" data-timezone="UTC">10:28PM - 23 Feb 21 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/Naheulf" target="_blank" rel="noopener">
          <img alt="Naheulf" src="https://user-images.githubusercontent.com/30189669/181076723-95e83d69-eabf-468a-9f5f-c1485deadb07.png" class="onebox-avatar-inline" width="20" height="20">
          Naheulf
        </a>
      </div>
    </div>

    <div class="labels">
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          books documentation
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          hammer Git
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          hammer GitHub
        </span>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">### Description of the Issue
The steps to setup the fork  for contribution don'<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">t works with a new fork and git up to date.

### Steps to Reproduce the Issue

1. Fork fantaisie-software/purebasic from the GitHub website
2. open Git Bash
3. Clone the repo with `git clone https://github.com/&lt;github_username&gt;/purebasic.git`
(Replace `&lt;github_username&gt;` with your own github username.)
4. Move to the repo usin `cd purebasic`
5. Follow steps from [CONTRIBUTING.md#setting-up-your-fork-for-contributions](https://github.com/fantaisie-software/purebasic/blob/c3bca303f9e66a4574b62e775eaf1b78ba4468aa/CONTRIBUTING.md#setting-up-your-fork-for-contributions)

### Expected Behavior
Should works.

### Actual Behavior 1
The second command generate the error: `fatal: 'upstream/devel' is not a commit and a branch 'devel' cannot be created from it`

### Fix 1
Run: `git fetch upstream` after the first command.

### Actual Behavior 2
After use the fix 1 the second command generate another error: `fatal: A branch named 'devel' already exists.`

### Fix 2
On the second command replace the `-b` (lower case B) with `-B` (uppercase B)

### More info
I let more experienced users check if theses fix are good or tricky. And I let english native speaker update the CONTRIBUTING.md file.</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by tajmone** [comment]
<aside class="quote group-github-employees" data-username="nyahbhinghiprincess" data-post="4" data-topic="184765">
<div class="title">
<div class="quote-controls"></div>
 nyahbhinghiprincess:</div>
<blockquote>
<p>Would be able to tell me if you see the same result?</p>
</blockquote>
</aside>
<p>I confirm, I can see the emojis correctly in you new card (also tested with in preview) and it works!</p>
<p>That was great, thanks for the improvement.</p>
